nova/src/utils.rs

```diff
@@ -35,10 +35,17 @@ pub fn iter_bits_le(bytes: &[u8]) -> impl Iterator<Item = bool> + '_ {
         .flat_map(|byte| (0..8).map(move |bit| ((1 << bit) & byte) != 0))
 }
 
+/// An implementation of base_2 log that will never fail, instead returning 0 for all values smaller than 1.
+/// It is 'safe' in the sense that it will not panic, unlike just calling `.unwrap()` on a `.checked_ilog2`.
+/// Usually used to compute the number of bits needed to index a set or a vector.
```

Comment:
It should mention that it returns ceil of log2 then, "safety" isn't the only difference from std ilog2, with a special case of `safe_log!(1) = 1`